setup.py

```diff
@@ -14,6 +14,17 @@
 
 import sys
 
+if sys.version_info < (3, 5):
+    error = """
+Matplotlib 3.0+ does not support Python 2.x, 3.0, 3.1, 3.2, 3.3, or 3.4.
+Beginning with Matplotlib 3.0, Python 3.5 and above is required.
+
+This may be due to an out of date pip.
+
+Make sure you have pip >= 9.0.1.
+"""
+    sys.exit(error)
```

Comment:
How does this behave if installed as a dependency of something via pip?  Modern pip should never try, but what do old versions of pip do?

Reply:
Well, that's why we error out and the message suggests that you install a new pip.

Reply:
In a normal install, it does this:
```
$ pip install -e .
Obtaining file:///.../matplotlib
    Complete output from command python setup.py egg_info:
    
    Matplotlib 3.0+ does not support Python 2.x, 3.0, 3.1, 3.2, 3.3, or 3.4.
    Beginning with Matplotlib 3.0, Python 3.5 and above is required.
    
    This may be due to an out of date pip.
    
    Make sure you have pip >= 9.0.1.
    
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in .../matplotlib/
```
I'm not sure how to test it out as a dependency, but I'd guess you get a similar error.